fgpyo/sam/__init__.py

```diff
@@ -189,6 +190,20 @@
 _IOClasses = (io.TextIOBase, io.BufferedIOBase, io.RawIOBase, io.IOBase)
 """The classes that should be treated as file-like classes"""
 
+SAM_UMI_DELIMITER: str = "-"
+"""Multiple UMI delimiter, which SAM specification recommends should be a hyphen;
+see specification here: https://samtools.github.io/hts-specs/SAMtags.pdf"""
+
+_VALID_UMI_CHARACTERS: Set[str] = set("ACGTN")
```

Comment:
can you move this method into a `fgpyo/vendor/illumina.py` file?  See comment from @tfenne here: https://github.com/fulcrumgenomics/fgpyo/pull/151#issuecomment-2192098876

Reply:
Just to clarify, `fgpyo/platform/illumina.py` is the destination we decided on